src/api-routes/persons/{osuId}/addresses.js

```diff
@@ -0,0 +1,29 @@
+import { serializeAddresses } from 'serializers/addresses-serializer';
+import { errorHandler } from 'errors/errors';
+import { getAddressesByOsuId } from 'db/oracledb/addresses-dao';
+
+/**
+ * Get addresses by OSU ID
+ *
+ * @type {RequestHandler}
+ */
+const get = async (req, res) => {
+  try {
+    const { query, params: { osuId } } = req;
+    const result = await getAddressesByOsuId(osuId, query);
+    const serializedAddresses = serializeAddresses(result, query, osuId);
```

Comment:
We only wanna return an empty array if the person exists but no address records are found. If a person doesn't exist, we should return 404.

You can execute a query to do a simple check before running the real query like this: https://github.sig.oregonstate.edu/ecs-data/persons-api-contrib/blob/2cab8153eb93a18d35ecbd22f51c18668ae7d12b/AbstractPersonsDAO.groovy#L4-L9

Same thing for all other sub-resource endpoints.